### PR TITLE
Add CMake support

### PR DESCRIPTION
**Pull request by jchristopherson**
Hi @gha3mi.  I have situations in my current workflow that dictate the use of CMake vs. FPM.  I'm sure there are others as well. To that end, please consider this addition of CMake as a build system.  I've also include CI specific to the CMake build to ensure its quality.
Thanks!

### CODE REVIEW

File: pnm_files/colors/black.ppm
Comment (gha3mi) [inline pnm_files/colors/black.ppm:1]: Hi @jchristopherson,

Thank you for sharing.

Before merging to main repo, could you kindly resend the changes without including the output files? I understand that it's not specified in the .gitignore. I intentionally included the outputs to showcase the results, and some of these results are used in the README file.

Comment (jchristopherson) [reply pnm_files/colors/black.ppm:1]: Hi @gha3mi.  Sorry about that.  All should be good now.

**Review by gha3mi — APPROVED** [review]

File: pnm_files/colors/black.ppm
Comment (gha3mi) [reply pnm_files/colors/black.ppm:1]: Thank you!
I will merge this into the main now.

Is it also possible to do the same for the examples directory?

Comment (jchristopherson) [reply pnm_files/colors/black.ppm:1]: You're welcome!  Thank you for allowing me to add the support for CMake!

Anyways, I will add support for the examples directory as well.  It shouldn't take me too long.

Comment (gha3mi) [reply pnm_files/colors/black.ppm:1]: It would be great! Then I will update the readme file.

**Pull request merged**